File: 38. Count and Say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    let res = "1";
    for (let i = 2; i <= n; i++)
    {
        temp = "";
        count = 1;
        for (let j = 1; j < res.length; j++)
        {
            if (res[j] != res[j-1])
            {
                temp += count.toString() + res[j-1];
                count = 0;
            }
            count += 1;
        }
        temp += count.toString() + res[res.length-1];
        res = temp;
    }
    return res;
};

File: 38. Count and Say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        res = "1"
        for i in xrange(2, n+1):
            count = 1
            temp = ""
            for j in xrange(1, len(res)):
                if res[j] != res[j-1]:
                    temp += str(count) + res[j-1]
                    count = 0
                count += 1
            temp += str(count) + res[len(res)-1]
            res = temp
        return res
